### PR TITLE
fixing hang when importing an invalid pmx template

### PR DESCRIPTION
**Pull request by dharmamike**

### CODE REVIEW

**Comment by rupakg** [comment]
:+1:

**Comment by alexwelch** [comment]
:+1:

File: app/scripts/services/pmx-converter.js [hunk 1]
```diff
@@ -10,20 +10,20 @@
...
+        images,
+        newJson = {};
 
       try {
         pmxJson = jsyaml.safeLoad(pmxYaml);
-      } catch (YamlException) {
-        return null;
+        images = pmxJson.images;
+        images.forEach(function (image) {
+          newJson[image.name.replace(' ', '')] = imageToServiceDefinition(image);
```
Comment (alexwelch) [inline app/scripts/services/pmx-converter.js]: So my recent hating on regexes is gonna make a hypocrite out of me, but I'll blame javascripts replace method instead... In this case you wouldn't catch multiple spaces:
```
'foo  bar'.replace(' ', '') >> "foo bar" // only replaces one space
'foo  bar'.replace(/\s+/, '') >> "foobar" // works
'foo  bar'.replace(/\s/g, '') >> "foobar" // also works
```

Comment (alexwelch) [inline app/scripts/services/pmx-converter.js]: I wish you could tell the replace method to be greedy... Maybe you can...

Comment (dharmamike) [inline app/scripts/services/pmx-converter.js]: Got it.  Apparently the global replace is not supported in chrome https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

**Comment by rupakg** [comment]
:+1:

**Pull request merged**